core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -445,10 +486,18 @@ private DayActivityDto createDayInactivity(UserAnonymizedDto userAnonymized, Goa
 				relevantGoals.stream().map(GoalDto::getGoalId).collect(Collectors.toSet()), interval.startDate, interval.endDate);
 	}
 
-	private Interval getInterval(LocalDate currentUnitDate, Pageable pageable, ChronoUnit timeUnit)
+	private Interval getInterval(LocalDate earliestPossibleDate, LocalDate currentUnitDate, Pageable pageable,
+			ChronoUnit timeUnit)
 	{
-		LocalDate startDate = currentUnitDate.minus(pageable.getOffset() + pageable.getPageSize() - 1L, timeUnit);
+		LocalDate requestedStartDate = currentUnitDate.minus(pageable.getOffset() + pageable.getPageSize() - 1L, timeUnit);
+		LocalDate startDate = TimeUtil.max(requestedStartDate, earliestPossibleDate);
+		if (timeUnit == ChronoUnit.WEEKS)
+		{
+			startDate = TimeUtil.getStartOfWeek(startDate);
+		}
 		LocalDate endDate = currentUnitDate.minus(pageable.getOffset() - 1L, timeUnit);
+		Require.that(!endDate.isBefore(startDate),
```

Comment:
`endDate.isBefore(earliestPossibleDate)` seems more logical

Reply:
It's certainly simpler, so I'll change it, but once you have seen it from the other side, the current construct seems more logical (at least to me :) ).